pkg/api/types.go

```diff
@@ -910,6 +910,55 @@ type ReplicationControllerList struct {
 	Items []ReplicationController `json:"items"`
 }
 
+// JobSpec is the specification of a job.
+// As the internal representation of a job, it may have either a TemplateRef or a Template set.
+type JobSpec struct {
+	// Completions is the number of desired successful completions of this job.
+	Completions int `json:"completions"`
+
+	// Selector is a label query over pods that should match the completed count.
+	Selector map[string]string `json:"selector"`
```

Comment:
Any guarantee given if half the count runs, the nodes are lost, and so count isn't tracked?  Or is the job controller expected to cache the pending completions until the job passes the threshold?  Can completed pods not be deleted until enough job pods have run?  Maybe i missed the deeper discussion on this...